source/server/listener_manager_impl.cc

```diff
@@ -223,21 +223,37 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, const std::st
     addListenSocketOptions(
         Network::SocketOptionFactory::buildLiteralOptions(config.socket_options()));
   }
+  bool is_quic{false};
```

Comment:
The fact that we have QUIC as an extension in general seems to be placing a huge amount of contortions on the implementation.

I'm going to be honest in saying that I find it extremely unlikely that there are going to be multiple QUIC implementations. I'm wondering if this should become core code much like the H1/H2 codecs? Any thoughts here? If we keep this as an extension, how do we keep this kind of leakage in this code and how would we differentiate between different QUIC implementations?

Reply:
Okay, I modified the name to be "quiche_quic_listener" in the other PR and added isTransportConnectionless() to ActiveUdpListenerFactory to distinguish QUIC from RawUdp